app/controllers/api/v1/tasks_controller.rb

```diff
@@ -25,6 +32,30 @@ def update
 
   private
 
+  def restrict_user_to_have_one_active_task
+    if current_user.tasks.any?{|task| task.status == 'pending' || 'confirmed'}
+      render json: { error: "You already have an active task pending" }, status: 403
+      return
+    end
+  end
+
+  def render_error_message(task)
+    case
+    when task.task_items.count >= 40
+      message = "You have to many products selected."
+    when task.task_items.count < 5
+      message = "You have to pick at least 5 products."
+    else
+      message = "Internal problem. Contact support. No activity specified"
```

Comment:
We are experiencing internal errors. Please refresh the page and contact support.

Reply:
done